### PR TITLE
[jsk_2016_01_baxter_apc] New eus robot model

### PR DESCRIPTION
**Pull request by pazeshun**
screenshot from 2016-04-16 09 43 54
The new gripper is added to the right arm. The end-coords of the right arm is associated with the vacuum pad of the gripper. We can rotate the joint of the gripper from Euslisp.
screenshot from 2016-04-16 09 54 27
screenshot from 2016-04-16 09 51 04
If #1296 is merged, I will rebase origin/master and erase WIP tag of this PR.

### CODE REVIEW

**Comment by pazeshun** [comment]
Please merge this.

**Comment by pazeshun** [comment]
@wkentaro I don't know why travis failed. I couldn't find errors in the log of travis. Please help.

**Comment by k-okada** [comment]
```
[jsk_2016_01_baxter_apc:make]     pkg_path = rp.get_path(args[0])



                                                                                
[jsk_2016_01_baxter_apc:make]   File "/usr/lib/python2.7/dist-packages/rospkg/rospack.py", line 200, in get_path




                                                                                
[jsk_2016_01_baxter_apc:make]     raise ResourceNotFound(name, ros_paths=self._ros_paths)




                                                                                
[jsk_2016_01_baxter_apc:make] rospkg.common.ResourceNotFound: softkinetic_camera



                                                                                
[jsk_2016_01_baxter_apc:make] ROS path [0]=/opt/ros/indigo/share/ros



                                                                                
[jsk_2016_01_baxter_apc:make] ROS path [1]=/workspace/ros/ws_jsk_apc/src/jsk_apc/jsk_2015_05_baxter_apc



                                                                                
[jsk_2016_01_baxter_apc:make] ROS path [2]=/workspace/ros/ws_jsk_apc/src/jsk_apc/jsk_apc2015_common
```
みたい．softkinetic_cameraの変更はもうmasterに含まれているか？というのと，
softkineticはどこから落としてきているか ? apt ? source ?

**Comment by pazeshun** [comment]
sourceです。fc.rosinstallでインストールされると思います。
ただ、こういうエラーは前から出てたので、無視してました。

**Comment by wkentaro** [comment]
catkin_tools 0.4.0がリリースされたようですね。
https://github.com/catkin/catkin_tools/releases

とりあえず、emptyコミットのPRを送るか、masterのビルドをリスタートしてみてください。

**Comment by k-okada** [comment]
right now, i have some trouble on new catkin ->
https://github.com/jsk-ros-pkg/jsk_travis/pull/243

--
◉ Kei Okada

**Comment by wkentaro-email-notification** [comment]
あと、travisの中ではfc.rosinstallは使われていないので、softkinetic_cameraはビルドされていないと思います。

--
和田 健太郎 / Kentaro Wada
http://wkentaro.com/

2016年4月19日 16:11 +0900 に Kentaro Wada<notifications@github.com>は書きました:

**Comment by pazeshun** [comment]
なるほど、じゃあ、Lispのモデルの自動生成がtravisでひっかかりますね。
どうするのが一番いいのでしょうか

**Comment by pazeshun** [comment]
This commit was test.  
There are two things I want to do.
1. Use xacro files in softkinetic_camera package as components of baxter.xacro
2. softkinetic_camera package depends on DepthSenseSDK, but we can' t get it automatically. So I want not to build softkinetic_camera on travis.

If there is no way to achieve these targets, I will copy meshes and xacros from softkinetic_camera to jsk_apc.

**Comment by pazeshun** [comment]
I copied meshes and xacros in the last commit.

**Comment by wkentaro** [comment]
@pazeshun Please add a license file like https://github.com/start-jsk/jsk_apc/blob/master/jsk_apc2015_common/LICENSE_berkeley_models .
It is BSD license so the content will be different. https://github.com/ipa320/softkinetic/blob/indigo_dev/softkinetic_camera/package.xml#L10

**Comment by knorth55** [comment]
https://github.com/start-jsk/jsk_apc/pull/1302/commits/bcb2e0d77ab23ab51d25a944aaa994ea4c1b268e this commit does not solve the travis build problem, does it?

**Comment by pazeshun** [comment]
@wkentaro Ok, I will add LICENSE.
@knorth55 Yes, commit bcb2e0d is no longer needed. I'll revise commits .

**Comment by pazeshun** [comment]
@wkentaro I realize that softkinetic_camera has no LICENSE file in meshes and xacros, so I don't know copyright holder's name and his organizarion.

**Comment by pazeshun** [comment]
@knorth55 fixed.

**Comment by pazeshun** [comment]
asus_xtion_pro_live.dae and asus_xtion_pro_live.png in softkinetic_camera maybe came from https://github.com/RobotnikAutomation/robotnik_sensors/tree/indigo-devel/meshes , but there is no copyright notice of robotnik_sensors package in softkinetic_camera. I am wondering there is no need to add LICENSE files

**Comment by wkentaro** [comment]
It is BSD License. https://github.com/ipa320/softkinetic/blob/indigo_dev/softkinetic_camera/package.xml#L10
So you can add License file like below:
```
Copyright © [YEAR] [COPYRIGHT OWNER]. All Rights Reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.

3. The name of the author may not be used to endorse or promote products derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY [LICENSOR] "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Reference: http://www.linfo.org/bsdlicense.html

**Comment by pazeshun** [comment]
@wkentaro The license is ok?

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/include/baxter-interface.l [hunk 1]
```diff
@@ -0,0 +1,287 @@
+;;
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/include/baxter-interface.l]: @pazeshun why this file is necessary? Is that impossible to reuse baxter-interface.l in jsk_2015_05_baxter_apc or baxtereus?

Comment (wkentaro) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/include/baxter-interface.l]: @pazeshun Please see: https://github.com/start-jsk/jsk_apc/pull/1365

**Pull request merged**